src/fia/fia/fia_state.rs

```diff
@@ -0,0 +1,131 @@
+use super::*;
+use crate::prelude::*;
+
+pub(super) struct FiaState<ID, Path, Product>
+where
+    ID: Hash,
+    Path: HasDerivationPath,
+    Product: HasHDPublicKey,
+{
+    phantom_id: PhantomData<ID>,
```

Comment:
`PhantomData` needed to retain the TYPE information (generics), needed until the actual impl of state includes all types.